crates/node/src/util.rs

```diff
@@ -1,14 +1,68 @@
 use std::{
     any::type_name,
     panic::{catch_unwind, AssertUnwindSafe},
+    sync::Mutex,
 };
 
 use anyhow::{anyhow, Context, Error};
 use napi::Status;
 use serde::de::DeserializeOwned;
 use swc::try_with_handler;
-use swc_common::{errors::Handler, sync::Lrc, SourceMap};
+use swc_common::{
+    errors::Handler,
+    sync::{Lrc, OnceCell},
+    SourceMap,
+};
+use tracing::instrument;
+use tracing_chrome::{ChromeLayerBuilder, FlushGuard};
+use tracing_subscriber::{
+    prelude::__tracing_subscriber_SubscriberExt, util::SubscriberInitExt, EnvFilter,
+};
+
+// Hold FlushGuard for the trace until it flushes out completely.
+#[derive(Default)]
+struct TraceFlushGuard {
+    inner: Option<FlushGuard>,
+}
+static FLUSH_GUARD: OnceCell<Mutex<TraceFlushGuard>> = OnceCell::new();
```

Comment:
I'm curious if there's better way to hold `FlushGuard` until a whole process lifecycle completes. We can't let this be dropped in `_register` or individual `transform*` interfaces.

Reply:
do we really need to hold gurad ? I remember I even didn't bind a ident to it

Reply:
Good q, per doc https://github.com/tokio-rs/tracing/blob/4365b3971b617784ca44a7be1f05358b2990056d/tracing-flame/src/lib.rs#L208 it says it'll flush writer when it's dropped.

The way I think is, napi-exported fns are not binded to process lifecycle and `trace_once_init()` will drop this guard if we don't hold it. At least, when I tried https://github.com/swc-project/swc/pull/3066 I confirmed it requires manual flush if I did not hold guard.

I think we need to ensure this won't be dropped in any way - just matter of how.